File: keyname-hash-generator/java/src/Hashing.java
import java.io.UnsupportedEncodingException;
import java.math.BigInteger;
import java.security.*;

public class Hashing {
	
	private static int numOfCharacters = 4;
	
	public static String hashKey(String key) throws UnsupportedEncodingException, NoSuchAlgorithmException {
		return Hashing.MD5(key);
	}
	
	public static String MD5(String key) throws UnsupportedEncodingException, NoSuchAlgorithmException {
		byte[] bytesOfMessage = key.getBytes("UTF-8");
		MessageDigest md = MessageDigest.getInstance("MD5");
		byte[] hashBytes = md.digest(bytesOfMessage);
		String hashString = new BigInteger(1,hashBytes).toString(16);
		hashString = hashString.substring(0, numOfCharacters);
		return(hashString+"/"+key);
	}
	
}

File: keyname-hash-generator/go/src/cloud/benchflow/commons/hashing/hashing.go
package hashing

import (
	"crypto/md5"
	"encoding/hex"
)

const numOfCharacters int = 4

func hashKey(key string) string {
	return MD5(key)
	}

func MD5(key string) string {
	hasher := md5.New()
    hasher.Write([]byte(key))
    hashString := hex.EncodeToString(hasher.Sum(nil))
	return (hashString[:numOfCharacters]+"/"+key)
	}